appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/NewYoungAndroidProjectWizard.java

```diff
@@ -43,7 +47,32 @@ public NewYoungAndroidProjectWizard() {
     // Initialize the UI
     setStylePrimaryName("ode-DialogBox");
 
-    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel());
+    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel(), new Validator() {
+      @Override
+      public boolean validate(String value) {
+        if(!value.matches("[A-Za-z][A-Za-z0-9_]*") && value.length() > 0) {
```

Comment:
I'm having trouble dealing with international characters. For this change it requires that I match unicode characters. The stack overflow post you linked me to gives two suggestions on how to do this. The first one requires me to import a couple classes from java.util.regex, but when I try to build app inventor after doing this, the build fails. The other suggestion includes the use of an embedded flag expression "(?U)". When I try this an internal error occurs whenever I type a character into the textbox. Any ideas?